### ISSUE TITLE
Auto-increment in application layer

### ISSUE THREAD

**Issue by GeertJohan** [go-gorp/gorp#271]
@fwang2002 commented on Feb 7, 2014 
    The value for auto increment field be set by application layer. 
    Now, the auto increment field can not be set, but sometimes it is needed.

**Comment by GeertJohan** [comment]
@fwang2002 could you please explain a bit more detailed what you mean by this? You want to set the auto-increment starting point on a table when it's created?

**Comment by fwang2002** [comment]
The auto-increment may start from a specified number, say 10000, thus reserved 9999 records for special use, which need the field to be set by application layer when inserting.
For example, the users table may reserve 10000 records which is for registering admin users, and the records starting from 10001 is for real end users.

**Comment by GeertJohan** [comment]
Right. Although I believe that reserving numbers for special users isn't really a good database design choice, I can see how it can be useful. I think a better example would be order numbers, where it would be nice if each number has the same length and so you start them at one million.
I think this could be added to the autoincrement tag; `db:"autoincrement:1000"`.
This issue will come up again after v2 was released (see #270).

**Comment by fwang2002** [comment]
If field `id` is an auto-increment field, and is starting by 10000. If the corresponding struct is 
type User struct {
    Id  int `db:"id"`
    ...
}
The following code can not run correctly.
dbMap.AddTableWithName(User{}, "users").SetKeys(true, "id")
var user = &User{Id: 9000}
dbMap.Insert(user)
after execution, the Id is reset to the value of database returned value. user.Id can not be set by application layer.
